util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,30 @@
+package com.twitter.util
+
+import java.util.logging.{Level, Logger}
+
+import scala.collection.mutable
+
+/** Provides a clean, lightweight interface for controlling a BatchExecutor
+  *
+  * @constructor create a new Batcher for a BatchExecutor
+  * @param executor the BatchExector to be used
+  */
+class Batcher[In, Out](
```

Comment:
We should make the constructor private[util] since folks outside of util won't have access to a `BatchExecutor` anyway.